src/bucket.ts

```diff
@@ -1339,7 +1343,11 @@ class Bucket extends ServiceObject {
 
     options = options || {};
 
-    const newLifecycleRules = arrify(rule).map(rule => {
+    if (!Array.isArray(rule)) {
+      rule = arrify(rule);
+    }
+
+    const newLifecycleRules = rule.map(rule => {
```

Comment:
I think `rules` might be clearer here (and we can drop `arrify`):

```suggestion
    const rules = Array.isArray(rule) ? rule : [rule];

    const newLifecycleRules = rules.map(rule => {
```